policy/policy.go

```diff
@@ -0,0 +1,406 @@
+// Copyright (c) 2016 Pani Networks
+// All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package policy
+
+import (
+	"fmt"
+	"github.com/romana/core/common"
+	"github.com/romana/core/tenant"
+	"log"
+	"strconv"
+	"strings"
+)
+
+// Policy provides Policy service.
+type PolicySvc struct {
+	client *common.RestClient
+	config common.ServiceConfig
+	store  policyStore
+}
+
+const (
+	infoListPath       = "/info"
+	findPath           = "/find"
+	policiesPath       = "/policies"
+	policyNameQueryVar = "policyName"
+)
+
+func (policy *PolicySvc) Routes() common.Routes {
+	routes := common.Routes{
+		common.Route{
+			Method:          "POST",
+			Pattern:         policiesPath,
+			Handler:         policy.addPolicy,
+			MakeMessage:     func() interface{} { return &common.Policy{} },
+			UseRequestToken: false,
+		},
+		common.Route{
+			Method:          "DELETE",
+			Pattern:         policiesPath,
+			Handler:         policy.deletePolicyHandler,
+			MakeMessage:     func() interface{} { return &common.Policy{} },
+			UseRequestToken: false,
+		},
+		common.Route{
+			Method:          "DELETE",
+			Pattern:         policiesPath + "/{policyID}",
+			Handler:         policy.deletePolicyHandler,
+			MakeMessage:     func() interface{} { return &common.Policy{} },
+			UseRequestToken: false,
+		},
+		common.Route{
+			Method:          "GET",
+			Pattern:         policiesPath,
+			Handler:         policy.listPolicies,
+			MakeMessage:     nil,
+			UseRequestToken: false,
+		},
+		common.Route{
+			Method:          "GET",
+			Pattern:         policiesPath + "/{policyID}",
+			Handler:         policy.getPolicy,
+			MakeMessage:     nil,
+			UseRequestToken: false,
+		},
+		common.Route{
+			Method:  "GET",
+			Pattern: findPath + policiesPath + "/{policyName}",
+			Handler: policy.findPolicyByName,
+		},
+	}
+	return routes
+}
+
+// augmentEndpoint augments the endpoint provided with appropriate information
```

Comment:
What a mysterious docstring. I can guess what happens, but I think this could be phrased more clearly.